loki/transform/transform_array_indexing.py

```diff
@@ -47,7 +54,14 @@ def shift_to_zero_indexing(routine):
                     # no shift for stop because Python ranges are [start, stop)
                     new_dims += [sym.RangeIndex((start, d.stop, d.step))]
                 else:
-                    new_dims += [d - sym.Literal(1)]
+                    shift = True
+                    for var in FindVariables().visit(d):
+                        if var in ignore:
+                            shift = False
+                    if shift:
+                        new_dims += [d - sym.Literal(1)]
+                    else:
+                        new_dims += [d]
```

Comment:
Just to propose a slightly streamlined alternative to avoid the explicit loop:
```suggestion
                    if ignore and any(var in ignore for var in FindVariables().visit(d)):
                        new_dims += [d]
                    else:
                        new_dims += [d - sym.Literal(1)]
```